server/users/methods.js

```diff
@@ -4,6 +4,22 @@ Meteor.methods({
     return Meteor.users.findOne({_id:userId},
                                 {fields: { emails: 0, services: 0, roles: 0, email: 0}});
   },
+
+  setUserProfile: function(profileInfo) {
+    var pattern = {
+      location: String,
+      bio: String,
+      website: String,
+      twitter: String,
+      github: String,
+      facebook: String,
+      linkedin: String
+    };
+    check(profileInfo, pattern);
+    Meteor.users.update({_id: Meteor.userId()},
```

Comment:
Before updating a record, confirm logged in user. 
and to update the profile object without overwriting, you can use dot-notation to access and set fields inside objects. `{$set: { profile.twitter: profile.twitter, profile.github:github ... }`